### DISCUSSION TITLE
Import data from .cfl (BART: MRI image reconstruction)

### DISCUSSION THREAD

**Discussion by jaykumar16** [Project-MONAI/MONAI#4503] [Ideas]
Hey, I am trying to implement the custom data load. May you please help me with this? I have the files that can read the data from .cfl and I want to implement this in monai.data.Dataset. 

Which files do I need to edit to read data from .cfl files?
if you tell me how to do this I can do this myself to save your time. 

Thank you for your help.

Here I am attaching the CFL read and write file, 
https://github.com/mrirecon/bart/blob/master/python/cfl.py

**Comment by Nic-Ma** [comment]
Hi @jaykumar16 ,

Thanks for your interest and contribution here.
I think you can develop a new image reader for the CFL files, refer to:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L358
And then people can use it as the `reader` arg of the `LoadImage` transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/io/array.py#L109
What do you think?

Thanks.

**Reply by jaykumar16** [reply]
I was able to write a function to read cfl files is there a process to merge this with monai GitHub.

**Reply by Nic-Ma** [reply]
Hi @jaykumar16 ,

Please refer to this doc:
https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md

Thanks.
